assets/src/edit-story/utils/getInUseFonts.js

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Returns a list of font names used across all pages in a story.
+ *
+ * @param {Array<import('../types').Page>} pages List of pages.
+ *
+ * @return {Array<string>} Font names.
+ */
+export function getInUseFontsForPages(pages) {
+  return [
+    ...new Set(
+      pages
+        .reduce((storyMemo, { elements = [] }) => {
+          return [
+            ...storyMemo,
+            ...elements.reduce((pageMemo, element) => {
+              if (element.type === 'text' && Boolean(element.font?.family)) {
+                return [...pageMemo, element.font?.family];
+              }
+              return pageMemo;
+            }, []),
+          ];
+        }, [])
+        .filter(Boolean)
+    ),
```

Comment:
Would this not be a bit more elegant with `.map()` and `.flat()` rather than `.reduce()` and all those spreads? I also think it'll be more performant, but that's probably on the order of nanos.

Reply:
Oh yeah. Good call, I sometimes forget about that ES19 gem, `.flat()`.